### PR TITLE
Registration API Cleaning

### PR DESCRIPTION
**Pull request by sbernard31**
Now we have a new store containing registration and observation, I cleaned the Registration API :

 - Renaming ClientRegistry by RegistrationService,
 - Removing internal API from the public interface.

(Next step probably doing the same thing for ObservationRegistry)

### CODE REVIEW

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/SecurityTest.java [hunk 1]
```diff
@@ -286,7 +286,7 @@ public void registered_device_with_x509cert_to_server_with_x509cert() throws Non
         helper.client.start();
         helper.waitForRegistration(1);
 
-        assertEquals(1, helper.server.getClientRegistry().allClients().size());
+        assertEquals(1, helper.server.getRegService().getAllRegistrations().size());
```
Comment (jvermillard) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/SecurityTest.java]: I wonder if we should try to be consistent here, one time we use "Reg" and another time "Registrations"

**Comment by jvermillard** [comment]
:+1:

**Comment by msangoi** [comment]
👍

**Comment by sbernard31** [comment]
Integrated in master (commits 417450051eccb129d7f05d2191ce101c5749e07d, 32623d8cdc3dc9fa561e3cba0102d9fcb00ffefb, a36f00e2bfdb3d9682f4bc3dfd89e09436673d54, 6cc8ccd43cf2986bc0ec3e77c5e2c32e116948d3, dc8b838a5ed7c02196c044897959214b2bf2165d)

**Pull request closed without merging**